tests/test_PixelFlags.py

```diff
@@ -103,28 +108,50 @@ def testOffimageNonfinite(self):
                                       np.array([False, True, True, True]))
         np.testing.assert_array_equal(catalog["base_PixelFlags_flag_edge"],
                                       np.array([False, True, True, True]))
+        np.testing.assert_array_equal(catalog["base_PixelFlags_flag_edgeCenter"],
+                                      np.array([False, True, True, True]))
+        np.testing.assert_array_equal(catalog["base_PixelFlags_flag_edgeCenterAll"],
+                                      np.array([False, True, True, True]))
 
     def testOffimage(self):
         """Test that sources at the boundary of the image get flag_offimage
-        and flag_edge set appropriately.
+        and flag_edge[Center[All]] set appropriately.
         """
         # These four will be explicitly set to values at the boundary; we
         # cannot add sources off the image in TestDataset.
         self.dataset.addSource(100000, lsst.geom.Point2D(20, 20))
         self.dataset.addSource(100000, lsst.geom.Point2D(20, 100))
-        self.dataset.addSource(100000, lsst.geom.Point2D(80, 100))
+        self.dataset.addSource(100000, lsst.geom.Point2D(40, 100))
         self.dataset.addSource(100000, lsst.geom.Point2D(80, 30))
+        # Half on the edge; no offimage or edgeCenterAll, but edge and edgeCenter.
+        self.dataset.addSource(100000, lsst.geom.Point2D(115, 30))
```

Comment:
Does "half on the edge" mean the source sits on the border of the edge region? Is the centroid _exactly_ on the border, or merely <1px off (I can see different actual/desired behavior for the two cases)?

Reply:
I've clarified the comments. It's about whether `edgeCenter` vs. `edgeCenterAll` is set. All that I care about is where central 3x3 region is, not the exact centroid.

Reply:
Yes, but the position of the 3×3 region will depend on where the centroid is, and by how much...

Reply:
Do the updated comments help?

Reply:
I guess? I'm not sure how the pixels can be "half" on the edge (a pixel is either flagged or not, no?)

Reply:
Ah, I meant "half of the" - I'll fix that.

Reply:
I meant that I'm not sure how you can get half of them like that -- I would have expected e.g. one row on and two off, or vice versa.

Reply:
Pedantism strikes again! You're right, but that doesn't matter for the test itself (nothing cares about the centroid itself, just the box around it).